docs/developing-contracts/examples/lottery.md

```diff
@@ -0,0 +1,226 @@
+
+## Structs
+
+The contract has the following structs:
+
+```rust
+struct Lottery {
+    lottery_state: LotteryState,
+    lottery_owner: ActorId,
+    token_address: Option<ActorId>,
+    players: BTreeMap<u32, Player>,
+    lottery_history: BTreeMap<u32, ActorId>,
+    lottery_id: u32,
+    lottery_balance: u128,
+}
+```
+where:
+
+`lottery_state` - Lottery state: Start Time, End time of the lottery
+
+`lottery_owner` - The address of the lottery owner who initialized the contract
+
+`token_address` - address of the token contract
+
+`players` - 'map' of the lottery players
+
+`lottery_history` - 'map'  of the lottery winners
+
+`lottery_id` – current lottery id
+
+`lottery_balance` - the total amount of bets in the lottery
+
+The `LotteryState` struct:
+
+```rust
+pub struct LotteryState {
+    pub lottery_started: bool,
+    pub lottery_start_time: u64,
+    pub lottery_duration: u64,
+}
+```
+
+The `Player` struct:
+
+```rust
+pub struct Player {
+    pub player_id: ActorId,
+    pub balance: u128,
+}
+```
+
+## Enums
+
+```rust
+pub enum LtAction {
+    Enter(u128),
+    StartLottery {
+        duration: u64,
+        token_address: Option<ActorId>,
+    },
+    LotteryState,
+    PickWinner,
+}
+
+pub enum LtEvent {
+    LotteryState(LotteryState),
+    Winner(u32),
+    PlayerAdded(u32),
+}
+
+pub enum LtState {
+    GetWinners,
+    GetPlayers,
+    BalanceOf(u32),
+    LotteryState,
+}
+
+pub enum LtStateReply {
+    Winners(BTreeMap<u32, ActorId>),
+    Players(BTreeMap<u32, Player>),
+    Balance(u128),
+    LotteryState(LotteryState),
+}
+```
+
+## Functions
+
+Lottery contract interacts with ERC20 contract through function `transfer_tokens`.
+
+```rust
+async fn transfer_tokens(
+	&mut self,
+	from: &ActorId, /// - the sender address
+	to: &ActorId, /// - the recipient address
+	amount_tokens: u128 /// - the amount of tokens
+)
+```
+
+This function sends a message (the action is defined in the enum `FTAction`) and gets a reply (the reply is defined in the enum `FTEvent`).
+
+```rust
+let _transfer_response: FTEvent = msg::send_and_wait_for_reply(
+    self.token_address.unwrap(), /// - the ERC-20 contracts address
+    FTAction::Transfer {		/// - action in ERC20-contract
+        from: *from,
+        to: *to,
+        amount: amount_tokens,
+    },
+    0,
+)
+```
+
+Launches a lottery. Only the owner can launch a lottery. Lottery must not have been launched earlier.
+
+```rust
+fn start_lottery(
+	&mut self,
+	duration: u64,
+	token_address: Option<ActorId>
+)
+```
+
+Called by a player in order to participate in the lottery. The player cannot enter the lottery more than once.
+
+```rust
+async fn enter(
+	&mut self,
+	amount: u128
+)
+```
+
+Lottery winner calculation. Only the owner can pick the winner.
+
+```rust
+async fn pick_winner(
+	&mut self
+)
+```
+
+These functions are called in `async fn main()` through enum `LtAction`.
```

Comment:
I think it also worth mentioning that it is the program entry point and the program expects a message in the format `LTAction`.

Reply:
Fixed on:
These functions are called in `async fn main()` through enum `LtAction`.
This is the entry point to the program, and the program is waiting for a message in `LtAction` format.